File: ruby/v2.6/campaigns/create_campaign.rb
#!/usr/bin/env ruby
# Encoding: utf-8
#
# Copyright:: Copyright 2016, Google Inc. All Rights Reserved.
#
# License:: Licensed under the Apache License, Version 2.0 (the "License");
#           you may not use this file except in compliance with the License.
#           You may obtain a copy of the License at
#
#           http://www.apache.org/licenses/LICENSE-2.0
#
#           Unless required by applicable law or agreed to in writing, software
#           distributed under the License is distributed on an "AS IS" BASIS,
#           WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
#           implied.
#           See the License for the specific language governing permissions and
#           limitations under the License.
#
# This example creates a campaign in a given advertiser.
#
# To create an advertiser, run create_advertiser.rb.

require_relative '../dfareporting_utils'
require 'securerandom'

def create_campaign(profile_id, advertiser_id)
  # Authenticate and initialize API service.
  service = DfareportingUtils.get_service()

  # Create a new campaign resource to insert.
  campaign = DfareportingUtils::API_NAMESPACE::Campaign.new({
    :advertiser_id => advertiser_id,
    :archived => false,
    :name => 'Example Campaign #%s' % SecureRandom.hex(3),
    :start_date => '2014-01-01',
    :end_date => '2020-01-01'
  })

  # Define a default landing page for the campaign.
  default_landing_page = DfareportingUtils::API_NAMESPACE::LandingPage.new({
    :name => 'Example Landing Page',
    :url => 'https://www.google.com'
  })

  # Insert the campaign.
  result = service.insert_campaign(profile_id, default_landing_page.name,
      default_landing_page.url, campaign)

  # Display results.
  puts 'Created campaign with ID %d and name "%s".' %
      [result.id, result.name]
end

if __FILE__ == $0
  # Retrieve command line arguments.
  args = DfareportingUtils.get_arguments(ARGV, :profile_id, :advertiser_id)

  create_campaign(args[:profile_id], args[:advertiser_id])
end

File: dotnet/v2.6/Campaigns/CreateCampaign.cs
/*
 * Copyright 2015 Google Inc
 *
 * Licensed under the Apache License, Version 2.0(the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
*/

using System;
using Google.Apis.Dfareporting.v2_6;
using Google.Apis.Dfareporting.v2_6.Data;

namespace DfaReporting.Samples {
  /// <summary>
  /// This example creates a campaign associated with a given advertiser. To
  /// create an advertiser, run CreateAdvertiser.cs.
  /// </summary>
  class CreateCampaign : SampleBase {
    /// <summary>
    /// Returns a description about the code example.
    /// </summary>
    public override string Description {
      get {
        return "This example creates a campaign associated with a given" +
            " advertiser. To create an advertiser, run CreateAdvertiser.cs.\n";
      }
    }

    /// <summary>
    /// Main method, to run this code example as a standalone application.
    /// </summary>
    /// <param name="args">The command line arguments.</param>
    public static void Main(string[] args) {
      SampleBase codeExample = new CreateCampaign();
      Console.WriteLine(codeExample.Description);
      codeExample.Run(DfaReportingFactory.getInstance());
    }

    /// <summary>
    /// Run the code example.
    /// </summary>
    /// <param name="service">An initialized Dfa Reporting service object
    /// </param>
    public override void Run(DfareportingService service) {
      long advertiserId = long.Parse(_T("INSERT_ADVERTISER_ID_HERE"));
      long profileId = long.Parse(_T("INSERT_USER_PROFILE_ID_HERE"));

      String campaignName = _T("INSERT_CAMPAIGN_NAME_HERE");
      String landingPageName = _T("INSERT_LANDING_PAGE_NAME_HERE");
      String url = _T("INSERT_LANDING_PAGE_URL_HERE");

      // Create the campaign structure.
      Campaign campaign = new Campaign();
      campaign.Name = campaignName;
      campaign.AdvertiserId = advertiserId;
      campaign.Archived = false;

      // Set the campaign start date. This example uses today's date.
      campaign.StartDate =
          DfaReportingDateConverterUtil.convertToDateString(DateTime.Now);

      // Set the campaign end date. This example uses one month from today's date.
      campaign.EndDate =
          DfaReportingDateConverterUtil.convertToDateString(DateTime.Now.AddMonths(1));

      // Insert the campaign.
      Campaign result =
          service.Campaigns.Insert(campaign, profileId, landingPageName, url).Execute();

      // Display the new campaign ID.
      Console.WriteLine("Campaign with ID {0} was created.", result.Id);
    }
  }
}
